packages/onchainkit/src/checkout/components/CheckoutProvider.tsx

```diff
@@ -322,15 +324,23 @@ export function CheckoutProvider({
       }
 
       // Open keys.coinbase.com for payment
-      await writeContractsAsync({
-        contracts: contractsRef.current,
-        capabilities: isSponsored
-          ? {
-              paymasterService: {
-                url: paymaster,
-              },
-            }
-          : undefined,
+      await sendCallsAsync({
+        calls: contractsRef.current.map((contract) => {
+          return {
+            to: contract.address,
+            abi: contract.abi,
+            functionName: contract.functionName,
+            args: contract.args,
+          };
+        }),
+        capabilities:
+          isSponsored && paymaster
+            ? {
+                paymasterService: {
+                  url: paymaster,
+                },
+              }
+            : undefined,
```

Comment:
Updated to the latest wagmi, so I had to update this to use `useSendCalls` instead of the experimental `useWriteContracts`